service/src/test/java/com/commercetools/util/HttpRequestUtilParallelTest.java

```diff
@@ -83,7 +141,20 @@ private void makeParallelRequests(final int nThreads, final int requests, final
                 .withFailMessage(format("The execution thread pool was not able to shutdown in time %s msec", maxCriticalPathDurationMsec))
```

Comment:
something maybe not related to this PR, but just a suggestion. Maybe move this threadpool shutdown in a separate method: 

````java
    private void assertThreadPoolIsShutdown(final int nThreads,
                                            final double requestsNumber,
                                            final ExecutorService newFixedThreadPool) throws InterruptedException {
        // length of the longest pipe in the multi thread pipeline,
        // literally it is an maximum integer number of sequential requests in one pipe (thread)
        final int criticalPathLength = (int) Math.ceil(requestsNumber / nThreads);

        // longest expected time of one successful request (even if retried),
        // which may have +RETRY_TIMES attempts additionally to the first (failed) attempt
        final int longestRequestTimeMsec = REQUEST_TIMEOUT * (1 + RETRY_TIMES);

        // await not more than (longestRequestTimeMsec * criticalPathLength) msec with
        // coefficient 1.5 is added to avoid test fails on some lags and threads switching timeouts.
        final long maxCriticalPathDurationMsec = (int) (1.5 * longestRequestTimeMsec * criticalPathLength);
        boolean interrupted = newFixedThreadPool.awaitTermination(maxCriticalPathDurationMsec, TimeUnit.MILLISECONDS);
        assertThat(interrupted)
                .withFailMessage(format("The execution thread pool was not able to shutdown in time %s msec", maxCriticalPathDurationMsec))
                .isTrue();
    }
````

and then in this method only -> 
````java
assertThreadPoolIsShutdown(nThreads, requestsNumber, newFixedThreadPool);
````

Reply:
well, don't think it is necessary right here:
the method is monolithic, not reusable, not public/exposed and fits one screen height. don't think it has much sense to split it .

Reply:
It was just a recommendation to make it maintainable and easier for future devs.